src/ucai/core/databricks.py

```diff
@@ -419,124 +388,122 @@ def _execute_uc_function(
-                statement=parametrized_statement.statement,
-                warehouse_id=self.warehouse_id,
-                parameters=parametrized_statement.parameters,
-                **execute_statement_args,  # type: ignore
+        parametrized_statement = get_execute_function_sql_stmt(function_info, parameters)
+        response = self.client.statement_execution.execute_statement(
+            statement=parametrized_statement.statement,
+            warehouse_id=self.warehouse_id,
+            parameters=parametrized_statement.parameters,
+            **execute_statement_args,  # type: ignore
+        )
+        # TODO: the first time the warehouse is invoked, it might take longer than
+        # expected, so it's still pending even after 6 times of retry;
+        # we should see if we can check the warehouse status before invocation, and
+        # increase the wait time if needed
+        if response.status and job_pending(response.status.state) and response.statement_id:
+            statement_id = response.statement_id
+            _logger.info("Retrying to get statement execution status...")
+            wait_time = 0
+            retry_cnt = 0
+            client_execution_timeout = int(
+                os.environ.get(
+                    UNITYCATALOG_AI_CLIENT_EXECUTION_TIMEOUT,
+                    DEFAULT_UC_AI_CLIENT_EXECUTION_TIMEOUT,
+                )
             )
-            # TODO: the first time the warehouse is invoked, it might take longer than
-            # expected, so it's still pending even after 6 times of retry;
-            # we should see if we can check the warehouse status before invocation, and
-            # increase the wait time if needed
-            if response.status and job_pending(response.status.state) and response.statement_id:
-                statement_id = response.statement_id
-                _logger.info("Retrying to get statement execution status...")
-                wait_time = 0
-                retry_cnt = 0
-                client_execution_timeout = int(
-                    os.environ.get(
-                        UNITYCATALOG_AI_CLIENT_EXECUTION_TIMEOUT,
-                        DEFAULT_UC_AI_CLIENT_EXECUTION_TIMEOUT,
-                    )
+            while wait_time < client_execution_timeout:
+                wait = min(2**retry_cnt, client_execution_timeout - wait_time)
+                time.sleep(wait)
+                _logger.info(f"Retry times: {retry_cnt}")
+                response = self.client.statement_execution.get_statement(statement_id)
+                if response.status is None or not job_pending(response.status.state):
+                    break
+                wait_time += wait
+                retry_cnt += 1
+            if response.status and job_pending(response.status.state):
+                return FunctionExecutionResult(
+                    error=f"Statement execution is still pending after {wait_time} "
```

Comment:
```suggestion
                    error=f"Statement execution is still {response.status.state} after {wait_time} "
```

can we include the status?